### PR TITLE
NETOBSERV-674 : Added management of prometheus rule

### PR DESCRIPTION
**Pull request by OlivierCazade**
This PR add support for PrometheusRule to configure different alerts.

I also added change detection to service monitor which for now, once created, is not checked.

### CODE REVIEW

File: controllers/flowlogspipeline/flp_common_objects.go [hunk 1]
```diff
@@ -576,3 +578,47 @@ func (b *builder) serviceMonitor() *monitoringv1.ServiceMonitor {
...
+		},
+		Spec: monitoringv1.PrometheusRuleSpec{
+			Groups: []monitoringv1.RuleGroup{
+				{
+					Name: "NetobservFlowLogsPipeline",
+					Rules: []monitoringv1.Rule{
+						{
+							Alert: "NetobservNoFlows",
+							Annotations: map[string]string{
+								"description": "Netobserv flowlogs-pipeline is not receiving any flow, this is either a connection issue with the agent, or an agent issue",
```
Comment (jotak) [inline controllers/flowlogspipeline/flp_common_objects.go:597]: Can you replace `Netobserv` with `NetObserv` (capital O) ? There's the occurrence in several places in this file

**Comment by OlivierCazade** [comment]
@jotak since we are still testing how changes in the CRD will be handle, do you mind if I create a following task to add an ignore list entry in the CRD?

**Review by jotak — APPROVED** [review]

**Comment by jotak** [comment]
@OlivierCazade can you create that follow-up task?

**Comment by OlivierCazade** [comment]
@jotak done: https://issues.redhat.com/browse/NETOBSERV-881

**Pull request merged**